main.cpp

```diff
@@ -20,29 +10,37 @@ vector<string> split(string path) {
 		directories.push_back(element);
 		path = path.substr(cut + 1, path.size());
 	}
-
 	return directories;
 }
 
 int main() {
-	cout << "Welcome in the terminal. Press 'q' to exit." << endl;
+	cout << "Please add a valid file name to load filesystem from. " << endl;
+	cout << "Welcome in the terminal. Write exit' to exit." << endl;
 	string CurrentFolder = "root";
 	Dictionary d;
 	string parancs;
-
-	while (parancs != "q") {
+	string dirname;
+	string fsname;
+	cin >> fsname;
+	while (d.validcommand(fsname)) {
+		cout << "Not valid name. Please add a valid file name to load filesystem from. " << endl;
+		cin >> fsname;
+		if (fsname == "exit") break;
+	}
```

Comment:
Na, egyreszt ket kiiratasod isvan, az egyik ker egy filenevet, a masik meg mar parancsokat var. 
De aztan a fajlnevet olvasod be eloszor, amirol leteszteled, hogy valid command-e? (minek?!)
Illetve ez typic helye egy `do while` ciklusnak, amikor ilyet akarunk :-)

Reply:
A második kiiratás az igazából egy ilyen knowhow hogy hogyan lehet exitelni, nem vár még ott parancsot. Az első kiiratás pedig arról szól, hogy adjon meg valid fájlnevet. A validcommanddel azt szűröm ki, hogy az input ne lehessen valami parancs, mert először fájlnév kell. De a szöveg ezt nem feltétlen impikálja, szóval javítottam.

Reply:
Oke, igy ertem, meg annyi, hogy talan logikusabb lenne az a ket kiiratas akkor forditott sorrendben.De ez mar csak nitpicking.